### PR TITLE
Fix E0132 error display

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Error and note now use the same span.

r? @jonathandturner

### CODE REVIEW

File: src/librustc_typeck/lib.rs [hunk 1]
```diff
@@ -261,9 +261,10 @@ fn check_start_fn_ty(ccx: &CrateCtxt,
                     match it.node {
                         hir::ItemFn(_,_,_,_,ref ps,_)
                         if ps.is_parameterized() => {
-                            struct_span_err!(tcx.sess, start_span, E0132,
+                            let sp = ps.span().unwrap();
```
Comment (jonathandturner) [inline src/librustc_typeck/lib.rs]: This looks okay, though I'm not 100% sure there's always a span if it's .is_parameterized(). 

We could support a fallback with something like:

```rust
let sp = if let Some(span) = ps.span() { span } else { start_span };
```

**Comment by jonathandturner** [comment]
Looks okay, though I left a comment on one way we might allow it to fallback to the larger span if there's ever a case we don't have the smaller span.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
That is quite the strange error!

**Comment by alexcrichton** [comment]
@bors: retry

* not sure why...

**Pull request merged**